server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/GlobalOrdinalValuesSource.java

```diff
@@ -53,10 +73,30 @@ class GlobalOrdinalValuesSource extends SingleDimensionValuesSource<BytesRef> {
         int reverseMul
     ) {
         super(bigArrays, format, type, missingBucket, missingOrder, size, reverseMul);
+        this.uniqueValueCount = uniqueValueCount;
         this.docValuesFunc = docValuesFunc;
         this.values = bigArrays.newLongArray(Math.min(size, 100), false);
     }
 
+    /**
+     * Return the number of unique values of this source. Note that some unique
+     * values might not produce buckets if the query doesn't match documents
+     * that contain these values.
+     */
+    long getUniqueValueCount() {
+        return uniqueValueCount;
+    }
+
+    /**
+     * Return whether the source can be used for dynamic pruning.
+     */
+    boolean mayDynamicallyPrune() {
+        // If missing bucket is requested we have no way to tell lucene to efficiently match
+        // buckets in a range AND missing values.
+        // If the field type is IP, we cannot compare them as keywords.
+        return missingBucket == false && fieldType instanceof IpFieldMapper.IpFieldType == false;
```

Comment:
The check on the field type looks too fragile to me: if we were to introduced a new field type that also uses ordinals, then we'd probably miss updating this exclusion list. I would rather check if the index has terms  like what is done here: https://github.com/elastic/elasticsearch/pull/92060/files#diff-f61bc9e8ad11882e081009eb8e86945004041b3326497cec8672414645268dafR207-R208. Maybe add the IndexReader to this method signature so that it can check the content of the index and propagate it from `CompositeAggregator` which should have access to it via the `AggregationContext`.